js/modules/k6/experimental/tracing/client_test.go

```diff
@@ -0,0 +1,159 @@
+package tracing
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/dop251/goja"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.k6.io/k6/js/modulestest"
+)
+
+func TestClientInstrumentArguments(t *testing.T) {
+	t.Parallel()
+
+	// Although the traceparent header is case insensitive, the
+	// Go http.Header sets it capitalized.
+	traceparentHeaderName := "Traceparent"
+	testTraceID := "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-00"
+
+	t.Run("no args should fail", func(t *testing.T) {
+		t.Parallel()
+
+		testSetup := modulestest.NewRuntime(t)
+		client := Client{vu: testSetup.VU}
+		traceContextHeader := http.Header{}
+		traceContextHeader.Add(traceparentHeaderName, testTraceID)
+
+		_, err := client.instrumentArguments(traceContextHeader)
+		require.Error(t, err)
+	})
+
+	t.Run("1 arg should initialize params successfully", func(t *testing.T) {
+		t.Parallel()
+
+		testSetup := modulestest.NewRuntime(t)
+		rt := testSetup.VU.Runtime()
+		client := Client{vu: testSetup.VU}
+		traceContextHeader := http.Header{}
+		traceContextHeader.Add(traceparentHeaderName, testTraceID)
+
+		gotArgs, gotErr := client.instrumentArguments(traceContextHeader, goja.Null())
```

Comment:
This seems to repeaat in almost all of the cases so I would  get it out. Maybe in a struct that has the fields that I am using instead of returning multiple results, most of which are not used. See https://github.com/grafana/k6/pull/2845/commits/3aedeedfb42ade8994a286d7756723c4b75b3f78 for a variant of what I am talking about

Reply:
Initially, I wanted to make table tests, but it appeared hard, if possible to do it in a comprehensible/maintainable way. Your solution is a good tradeoff indeed, and I applied it 👍🏻